chromium_src/chrome/browser/ui/views/javascript_tab_modal_dialog_view_views.cc

```diff
@@ -0,0 +1,12 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/ui/javascript_dialogs/javascript_tab_modal_dialog_manager_delegate_desktop.h"
+
+#define CreateNewDialog CreateNewDialog_ChromiumImpl
```

Comment:
Maybe it would be good to have a comment here indicating why this substitution is necessary.